pkg/controller/service/reconciler.go

```diff
@@ -66,29 +68,57 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	log.Info("Reconciling service")
 
 	var (
-		ip    string
-		patch = client.MergeFrom(service.DeepCopy())
+		ips      []string
+		nodePort int32
+		patch    = client.MergeFrom(service.DeepCopy())
 	)
 
-	for i, servicePort := range service.Spec.Ports {
-		switch {
-		case (key == keyIstioIngressGateway || key == keyVirtualGardenIstioIngressGateway) && servicePort.Name == "tcp":
-			service.Spec.Ports[i].NodePort = nodePortIstioIngressGateway
-			if r.IsMultiZone {
-				// Docker desktop for mac v4.23 breaks traffic going through a port mapping to a different docker container.
-				// Setting external traffic policy to local mitigates the issue for multi-node setups, e.g. for gardener-operator.
-				service.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicyLocal
+	switch {
+	case key == keyIstioIngressGateway:
+		nodePort = nodePortIstioIngressGateway
+		if r.IsMultiZone {
+			// Docker desktop for mac v4.23 breaks traffic going through a port mapping to a different docker container.
+			// Setting external traffic policy to local mitigates the issue for multi-node setups, e.g. for gardener-operator.
+			service.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicyLocal
+		}
+		if err := r.Client.Get(ctx, keyVirtualGardenIstioIngressGateway, ptr.To(corev1.Service{})); err != nil {
+			if apierrors.IsNotFound(err) {
+				if nodes, err := r.getNodesInternalIPs(ctx); err != nil {
+					return reconcile.Result{}, err
+				} else {
+					ips = append(ips, nodes...)
+				}
```

Comment:
Drop the `else` after an `if` block which always returns.

```suggestion
				}
				ips = append(ips, nodes...)
```